Directions Example/ContentView.swift

```diff
@@ -0,0 +1,135 @@
+        return travelTimeFormatter.string(from: route.expectedTravelTime)!
+    }
+
+    func formattedTypicalTravelTime(for route: Route) -> String {
+        if let typicalTravelTime = route.typicalTravelTime,
+           let formattedTypicalTravelTime = travelTimeFormatter.string(from: typicalTravelTime) {
+            return formattedTypicalTravelTime
+        }
+        else {
+            return "Not available"
+        }
+    }
+
+    func stepDescriptions(for step: RouteStep) -> String {
+        var description: String = ""
+        let direction = step.maneuverDirection?.rawValue ?? "none"
+        description.append("\(step.instructions) [\(step.maneuverType) \(direction)]")
+        if step.distance > 0 {
+            let formattedDistance = distanceFormatter.string(fromMeters: step.distance)
+            description.append(" (\(step.transportType) for \(formattedDistance))")
+        }
+        return description
+    }
+}
+
+struct ContentView: View {
+    @ObservedObject
+    var vm: DirectionsViewModel
+
+    var body: some View {
+        ScrollView {
+            LazyVStack(spacing: 10, content: {
+                ForEach(vm.routes, id: \.routeIdentifier) { route in
+                    VStack(alignment: .leading, spacing: 3) {
+                        headerView(for: route)
+                        ForEach(0..<route.legs.count, id: \.self) { legIdx in
+                            if let source = route.legs[legIdx].source?.name,
+                               let destination = route.legs[legIdx].destination?.name {
+                                Text("From '\(source)' to '\(destination)'").font(.title2)
+                            }
+                            else {
+                                Text("Steps:").font(.title2)
+                            }
+                            stepsView(for: route.legs[legIdx])
+                        }
+                    }
+                }
+            })
+        }
+        .padding(5)
+        .onAppear { vm.loadRoutes() }
+    }
+
+    @ViewBuilder
+    private func headerView(for route: Route) -> some View {
+        VStack(alignment: .leading, spacing: 10) {
+            HStack {
+                Text("Route: ").fontWeight(.bold)
+                Text(route.description)
+                    .fixedSize(horizontal: false, vertical: true)
+            }
+            HStack {
+                Text("Distance: ").fontWeight(.bold)
+                Text(vm.formattedDistance(for:route))
+            }
+            HStack {
+                Text("ETA: ").fontWeight(.bold)
+                Text(vm.formattedTravelTime(for: route))
+            }
+            HStack {
+                Text("Typical travel time: ").fontWeight(.bold)
+                Text(vm.formattedTypicalTravelTime(for: route))
+            }
+            Divider()
+        }
+    }
+
+    @ViewBuilder
+    private func stepsView(for leg: RouteLeg) -> some View {
```

Comment:
This is a slick UI compared to the former console output. 😅 In the future, we could make it so that long-pressing a steps view copies the contents, making it easier to share a step as part of developer feedback. Or tapping a steps view could even push a preview of the step that consists of the step’s `shape` plotted against a map using MapboxStatic.swift. But this is more than enough to tide us over for now. 💯

Reply:
Thank you for ideas! I have a few on my own like make the example more interactive, but I will tackle this separately. 
I will create a task with a new set of improvements to gather the ideas.